### PR TITLE
V0.3.0

### PR DESCRIPTION
**Pull request by aldemirenes**

### CODE REVIEW

**Comment by alexanderdean** [comment]
Thanks @aldemirenes ! Review from @jbeemster sounds like a great next step.

File: Vagrantfile [hunk 1]
```diff
@@ -3,25 +3,31 @@ Vagrant.configure("2") do |config|
   config.vm.box = "ubuntu/trusty64"
   config.vm.hostname = "snowplow-mini"
   config.ssh.forward_agent = true
+  # Set the name of the VM. See: http://stackoverflow.com/a/17864388/100134
+  #config.vm.define :snowplowmini do |snowplowmini|
+  #end
```
Comment (jbeemster) [inline Vagrantfile]: If this section is not being used can it be removed?

File: Vagrantfile [hunk 2]
```diff
@@ -3,25 +3,31 @@ Vagrant.configure("2") do |config|
...
   config.vm.network "forwarded_port", guest: 5601, host: 5601
   config.vm.network "forwarded_port", guest: 8081, host: 8081
+  config.vm.network "forwarded_port", guest: 10000, host: 10000
 
   config.vm.provider :virtualbox do |vb|
     vb.name = Dir.pwd().split("/")[-1] + "-" + Time.now.to_f.to_i.to_s
     vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
     vb.customize [ "guestproperty", "set", :id, "--timesync-threshold", 10000 ]
-    vb.memory = 4096
+    vb.memory = 4096
```
Comment (jbeemster) [inline Vagrantfile]: Is there a reason for adding whitespace at the end of this line?

File: Vagrantfile [hunk 3]
```diff
@@ -3,25 +3,31 @@ Vagrant.configure("2") do |config|
...
   config.vm.provider :virtualbox do |vb|
     vb.name = Dir.pwd().split("/")[-1] + "-" + Time.now.to_f.to_i.to_s
     vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
     vb.customize [ "guestproperty", "set", :id, "--timesync-threshold", 10000 ]
-    vb.memory = 4096
+    vb.memory = 4096 
     vb.cpus = 1
   end
 
+
```
Comment (jbeemster) [inline Vagrantfile]: Extra whitespace?

File: provisioning/resources/configs/enrichments/campaign_attribution.json [hunk 4]
```diff
@@ -0,0 +1,20 @@
+{
+	"schema": "iglu:com.snowplowanalytics.snowplow/campaign_attribution/jsonschema/1-0-1",
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/campaign_attribution.json]: Whitespace

File: provisioning/resources/configs/enrichments/campaign_attribution.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+	"data": {
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/campaign_attribution.json]: Whitespace

File: provisioning/resources/configs/enrichments/event_fingerprint_config.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+    "schema": "iglu:com.snowplowanalytics.snowplow/event_fingerprint_config/jsonschema/1-0-0",
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/event_fingerprint_config.json]: Whitespace

File: provisioning/resources/configs/enrichments/event_fingerprint_config.json [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+    "data": {
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/event_fingerprint_config.json]: Whitespace

File: provisioning/resources/configs/enrichments/ip_lookups.json [hunk 8]
```diff
@@ -0,0 +1,16 @@
+{
+        "schema": "iglu:com.snowplowanalytics.snowplow/ip_lookups/jsonschema/1-0-0",
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/ip_lookups.json]: Whitespace

File: provisioning/resources/configs/enrichments/ip_lookups.json [hunk 9]
```diff
@@ -0,0 +1,16 @@
...
+        "data": {
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/ip_lookups.json]: Whitespace

File: provisioning/resources/configs/enrichments/referer_parser.json [hunk 10]
```diff
@@ -0,0 +1,14 @@
+{
+	"schema": "iglu:com.snowplowanalytics.snowplow/referer_parser/jsonschema/1-0-0",
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/referer_parser.json]: Whitespace

File: provisioning/resources/configs/enrichments/referer_parser.json [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+	"data": {
+
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/referer_parser.json]: Whitespace

File: provisioning/resources/configs/enrichments/ua_parser_config.json [hunk 12]
```diff
@@ -0,0 +1,9 @@
+{
+    "schema": "iglu:com.snowplowanalytics.snowplow/ua_parser_config/jsonschema/1-0-0",
```
Comment (jbeemster) [inline provisioning/resources/configs/enrichments/ua_parser_config.json]: Inconsistent indenting between enrichments - please standardise on one!

File: provisioning/resources/configs/snowplow-elasticsearch-sink-bad.hocon [hunk 13]
```diff
@@ -66,13 +66,35 @@ sink {
...
+    }
+
+    cluster {
+      name: "elasticsearch"
+      index: "bad"
+      type: "bad"
+    }
   }
 
+
```
Comment (jbeemster) [inline provisioning/resources/configs/snowplow-elasticsearch-sink-bad.hocon]: Whitespace

File: provisioning/resources/control-plane/snowplow-mini-control-plane-api.go [hunk 14]
```diff
@@ -0,0 +1,35 @@
+package main
```
Comment (jbeemster) [inline provisioning/resources/control-plane/snowplow-mini-control-plane-api.go]: Missing license header

File: provisioning/resources/control-plane/snowplow-mini-control-plane-api.go [hunk 15]
```diff
@@ -0,0 +1,35 @@
...
+
+import (
+    "github.com/emicklei/go-restful"
+    "io"
+    "net/http"
+    "log"
+    "os/exec"
+    "flag"
+)
+
```
Comment (jbeemster) [inline provisioning/resources/control-plane/snowplow-mini-control-plane-api.go]: whitespace

File: provisioning/resources/init/caddy_init [hunk 16]
```diff
@@ -0,0 +1,99 @@
...
+# Short-Description: Start daemon at boot time
+# Description:       Enable service provided by daemon.
+### END INIT INFO
+
+dir="/usr/local/bin"
+
+cmd="caddy -conf /home/ubuntu/snowplow/configs/Caddyfile"
+user=""
+
+
```
Comment (jbeemster) [inline provisioning/resources/init/caddy_init]: whitespace

File: provisioning/resources/init/caddy_init [hunk 17]
```diff
@@ -0,0 +1,99 @@
...
+name="caddy_server"
+pid_file="/var/run/$name.pid"
+
+
```
Comment (jbeemster) [inline provisioning/resources/init/caddy_init]: whitespace

File: provisioning/resources/init/caddy_init [hunk 18]
```diff
@@ -0,0 +1,99 @@
...
+get_pid() {
+    cat "$pid_file"
+}
+
+is_running() {
+    [ -f "$pid_file" ] && ps `get_pid` > /dev/null 2>&1
+}
+
+
```
Comment (jbeemster) [inline provisioning/resources/init/caddy_init]: whitespace

File: provisioning/resources/init/snowplow_mini_control_plane_api [hunk 19]
```diff
@@ -0,0 +1,98 @@
...
+# Default-Stop:      0 1 6
+# Short-Description: Start daemon at boot time
+# Description:       Enable service provided by daemon.
+### END INIT INFO
+
+dir="/home/ubuntu/snowplow/bin/"
+cmd="./snowplow-mini-control-plane-api -scriptsPath=/home/ubuntu/snowplow/scripts/snowplow-mini-control-plane-api-scripts"
+user="ubuntu"
+
+
```
Comment (jbeemster) [inline provisioning/resources/init/snowplow_mini_control_plane_api]: whitespace

File: provisioning/resources/init/snowplow_mini_control_plane_api [hunk 20]
```diff
@@ -0,0 +1,98 @@
...
+get_pid() {
+    cat "$pid_file"
+}
+
+is_running() {
+    [ -f "$pid_file" ] && ps `get_pid` > /dev/null 2>&1
+}
+
+
```
Comment (jbeemster) [inline provisioning/resources/init/snowplow_mini_control_plane_api]: whitespace

File: provisioning/resources/ui/js/components/ControlPlane.tsx [hunk 21]
```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016 Snowplow Analytics Ltd. All rights reserved.
```
Comment (jbeemster) [inline provisioning/resources/ui/js/components/ControlPlane.tsx]: 2016-2017

File: provisioning/resources/ui/js/components/ControlPlane.tsx [hunk 22]
```diff
@@ -0,0 +1,47 @@
...
+import React = require("react");
+import ReactDOM = require("react-dom");
+import axios from 'axios';
+
+export class ControlPlane extends React.Component<{}, {}> {
+
+  public render() {
+    return (
+      <div className="tab-content">
+        <p>Press the buttons for controlling the Snowplow-Mini without ssh:<br></br><br></br>
```
Comment (jbeemster) [inline provisioning/resources/ui/js/components/ControlPlane.tsx]: > Press the buttons for controlling the Snowplow-Mini without ssh:

This feels like we are oversharing what it is doing.  Why does the user care that it is without ssh?

Can we change this to:

> The buttons below can be used to interact with the interact with the internal systems of Snowplow Mini:

Thoughts @alexanderdean ?

File: provisioning/resources/ui/js/components/ControlPlane.tsx [hunk 23]
```diff
@@ -0,0 +1,47 @@
...
+          <button type="button" title="Clear the cache for schemas" onClick={this.restartAllServices.bind(this) }>Restart all services</button>
```
Comment (jbeemster) [inline provisioning/resources/ui/js/components/ControlPlane.tsx]: > Clear the cache for schemas

Why is the title different to what is in the button text?

Can we add some small amount of usage information for _why_ the user would want to press that button?

File: provisioning/resources/ui/js/components/ControlPlane.tsx [hunk 24]
```diff
@@ -0,0 +1,47 @@
...
+
+  private restartAllServices(): void {
+    alert("Restarting all services...")
+
+    axios.put('/control-plane/restart-services', {}, {})
+      .then(function (response) {
+        alert("All services are restarted successfully")
+      })
+      .catch(function (error) {
+        alert("Error while restarting services")
```
Comment (jbeemster) [inline provisioning/resources/ui/js/components/ControlPlane.tsx]: What can the user do if this alert appears?  Is this calling for a hard refresh or replacement of the server?  Can we instruct the user what they should do in this case?

File: provisioning/roles/base/tasks/main.yml [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+    - libxslt1-dev 
+    - libxml2-dev
+    - zip
+    - libssl-dev
+    - unzip
+    - python-pip
+    - python-psycopg2
+    - apg
+    - mercurial
+    - git
```
Comment (jbeemster) [inline provisioning/roles/base/tasks/main.yml]: Are all of these packages required?

File: provisioning/roles/sp_mini_1_create_dirs_and_pipes/defaults/main.yml [hunk 26]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_1_create_dirs_and_pipes/defaults/main.yml]: whitespace

File: provisioning/roles/sp_mini_1_create_dirs_and_pipes/defaults/main.yml [hunk 27]
```diff
@@ -0,0 +1,15 @@
...
+executables_dir:      "{{main_dir}}/bin"
+unix_pipes_dir:       "{{main_dir}}/pipes"
+es_dir:               "{{main_dir}}/elasticsearch"
+scripts_dir:          "{{main_dir}}/scripts"
+init_dir:             "{{main_dir}}/init"
+ui_dir:               "{{main_dir}}/ui/dist"
+ansible_cache_dir:    "/var/cache/ansible" 
+software_install_dir: "/usr/local/share"
+user_binaries:        "/usr/local/bin"
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_1_create_dirs_and_pipes/defaults/main.yml]: whitespace

File: provisioning/roles/sp_mini_1_create_dirs_and_pipes/tasks/main.yml [hunk 28]
```diff
@@ -0,0 +1,51 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_1_create_dirs_and_pipes/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_1_create_dirs_and_pipes/tasks/main.yml [hunk 29]
```diff
@@ -0,0 +1,51 @@
...
+        src: "{{playbook_dir}}/resources/configs"
+        dest: "{{main_dir}}"
+        recursive: yes
+
+    - synchronize:
+        src: "{{playbook_dir}}/resources/init"
+        dest: "{{main_dir}}"
+        recursive: yes
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_1_create_dirs_and_pipes/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_2_install_postgresl/defaults/main.yml [hunk 30]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_2_install_postgresl/defaults/main.yml]: whitespace

File: provisioning/roles/sp_mini_2_install_postgresl/tasks/main.yml [hunk 31]
```diff
@@ -0,0 +1,54 @@
...
+
+- name: Install acl for creating Postgresql user
+  become: yes
+  apt:
+    name: "acl"
+    state: present
+    update_cache: yes
+    cache_valid_time: 3600
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_2_install_postgresl/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_2_install_postgresl/tasks/main.yml [hunk 32]
```diff
@@ -0,0 +1,54 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_2_install_postgresl/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_2_install_postgresl/defaults/main.yml [hunk 33]
```diff
@@ -0,0 +1,15 @@
...
+staging_dir:          "{{main_dir}}/staging"
+executables_dir:      "{{main_dir}}/bin"
+unix_pipes_dir:       "{{main_dir}}/pipes"
+es_dir:               "{{main_dir}}/elasticsearch"
+scripts_dir:          "{{main_dir}}/scripts"
+init_dir:             "{{main_dir}}/init"
+ui_dir:               "{{main_dir}}/ui/dist"
+ansible_cache_dir:    "/var/cache/ansible" 
+software_install_dir: "/usr/local/share"
+user_binaries:        "/usr/local/bin"
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_2_install_postgresl/defaults/main.yml]: Are all of these variables repeated in `sp_mini_1_create_dirs_and_pipes/defaults/main.yml`.  Is there a reason why these cannot be shared?

Or have these been included here by mistake?

File: provisioning/roles/sp_mini_2_install_postgresl/defaults/main.yml [hunk 34]
```diff
@@ -0,0 +1,15 @@
...
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_2_install_postgresl/defaults/main.yml]: whitespace

File: provisioning/roles/sp_mini_2_install_postgresl/tasks/main.yml [hunk 35]
```diff
@@ -0,0 +1,54 @@
...
+  apt:
+    name: "{{item}}"
+    state: latest
+  with_items:
+    - python-psycopg2
+  tags:
+    - postgresql
+    - db
+    - deps
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_2_install_postgresl/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/defaults/main.yml [hunk 36]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/defaults/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/defaults/main.yml [hunk 37]
```diff
@@ -0,0 +1,15 @@
...
+executables_dir:      "{{main_dir}}/bin"
+unix_pipes_dir:       "{{main_dir}}/pipes"
+es_dir:               "{{main_dir}}/elasticsearch"
+scripts_dir:          "{{main_dir}}/scripts"
+init_dir:             "{{main_dir}}/init"
+ui_dir:               "{{main_dir}}/ui/dist"
+ansible_cache_dir:    "/var/cache/ansible" 
+software_install_dir: "/usr/local/share"
+user_binaries:        "/usr/local/bin"
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/defaults/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/defaults/main.yml [hunk 38]
```diff
@@ -0,0 +1,15 @@
...
+staging_dir:          "{{main_dir}}/staging"
+executables_dir:      "{{main_dir}}/bin"
+unix_pipes_dir:       "{{main_dir}}/pipes"
+es_dir:               "{{main_dir}}/elasticsearch"
+scripts_dir:          "{{main_dir}}/scripts"
+init_dir:             "{{main_dir}}/init"
+ui_dir:               "{{main_dir}}/ui/dist"
+ansible_cache_dir:    "/var/cache/ansible" 
+software_install_dir: "/usr/local/share"
+user_binaries:        "/usr/local/bin"
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/defaults/main.yml]: This is the third instance of the exact same variables...

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 39]
```diff
@@ -0,0 +1,191 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 40]
```diff
@@ -0,0 +1,191 @@
...
+- name: Set version variables
+  set_fact:
+      kinesis_package: 'snowplow_kinesis_r85_metamorphosis.zip'
+      iglu_server_package: 'iglu_server_0.2.0.zip'
+      kibana_v: '4.0.1'
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 41]
```diff
@@ -0,0 +1,191 @@
...
+    update_cache: yes
+    cache_valid_time: 3600 
+
+- name: Add Java 8 repository
+  become: yes
+  apt_repository: 
+    repo: 'ppa:webupd8team/java'
+    state: present
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 42]
```diff
@@ -0,0 +1,191 @@
...
+- name: Signed Oracle License
+  become: yes
+  shell: "echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | sudo /usr/bin/debconf-set-selections"
+  register: oracle_license_signed
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 43]
```diff
@@ -0,0 +1,191 @@
...
+- name: Install GO 1.7
+  become: yes 
+  block:
+    - shell: "wget https://storage.googleapis.com/golang/go1.7.3.linux-amd64.tar.gz -P {{staging_dir}}"
+    - shell: "tar -C /usr/local -xzf {{staging_dir}}/go1.7.3.linux-amd64.tar.gz"
+    - file:
+        path: "~/go"
+        state: directory
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 44]
```diff
@@ -0,0 +1,191 @@
...
+  synchronize:
+    src: "{{playbook_dir}}/resources/control-plane/scripts/"
+    dest: "{{scripts_dir}}/snowplow-mini-control-plane-api-scripts"
+
+- name: Check Kinesis Packages 
+  stat:
+    path: "{{staging_dir}}/{{kinesis_package}}"
+  register: check_kinesis_packages_result
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 45]
```diff
@@ -0,0 +1,191 @@
...
+- name: Download Kinesis Packages
+  get_url: 
+    url: "http://dl.bintray.com/snowplow/snowplow-generic/{{kinesis_package}}"
+    dest: "{{staging_dir}}"
+  when: check_kinesis_packages_result.stat.exists == False
+  register: kinesis_packages_downloaded
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 46]
```diff
@@ -0,0 +1,191 @@
...
+- name: Unzip downloaded Kinesis Packages
+  shell: "unzip {{staging_dir}}/{{kinesis_package}} -d {{executables_dir}}"
+  when: kinesis_packages_downloaded|changed
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 47]
```diff
@@ -0,0 +1,191 @@
...
+- name: Check Iglu Server
+  stat:
+    path: "{{staging_dir}}/{{iglu_server_package}}"
+  register: check_iglu_server_result
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 48]
```diff
@@ -0,0 +1,191 @@
...
+- name: Download Iglu Server
+  get_url: 
+    url: "http://bintray.com/artifact/download/snowplow/snowplow-generic/{{iglu_server_package}}"
+    dest: "{{staging_dir}}"
+  when: check_iglu_server_result.stat.exists == False
+  register: iglu_server_downloaded
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 49]
```diff
@@ -0,0 +1,191 @@
...
+- name: Install Elasticsearch Head Plugin
+  become: yes
+  shell: "/usr/share/elasticsearch/bin/plugin --install mobz/elasticsearch-head"
+
+- name: Check Kibana
+  stat:
+    path: "{{staging_dir}}/{{kinesis_package}}"
+  register: check_kibana_result
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 50]
```diff
@@ -0,0 +1,191 @@
...
+- name: Download Kibana
+  get_url: 
+    url: "https://download.elasticsearch.org/kibana/kibana/kibana-{{kibana_v}}-linux-x64.zip"
+    dest: "{{staging_dir}}"
+  when: check_kinesis_packages_result.stat.exists == False
+  register: kibana_downloaded
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 51]
```diff
@@ -0,0 +1,191 @@
...
+- name: Unzip downloaded Kibana package
+  become: yes
+  shell: "unzip {{staging_dir}}/kibana-{{kibana_v}}-linux-x64.zip -d /opt/"
+  when: kibana_downloaded|changed
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml [hunk 52]
```diff
@@ -0,0 +1,191 @@
...
+- name: Symlink for kibana
+  become: yes
+  file: 
+    src: "/opt/kibana-{{kibana_v}}-linux-x64"
+    dest: "/opt/kibana"
+    state: link
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_3_setup_apps/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_4_build_ui/defaults/main.yml [hunk 53]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_4_build_ui/defaults/main.yml]: This is the fourth version of the same vars!

File: provisioning/roles/sp_mini_4_build_ui/tasks/main.yml [hunk 54]
```diff
@@ -0,0 +1,11 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_4_build_ui/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_4_build_ui/tasks/main.yml [hunk 55]
```diff
@@ -0,0 +1,11 @@
...
+- name: Install npm packages based on package.json.
+  become: yes
+  npm:
+    path: "{{playbook_dir}}/resources/ui"
+
+- name: Create dist files
+  become: yes
+  shell: "cd {{playbook_dir}}/resources/ui; tsc -p js --outDir dist/ && browserify dist/SnowplowMiniApp.js -o dist/bundle.js && uglifyjs dist/bundle.js > dist/snowplow-mini.js"
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_4_build_ui/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_5_copy_ui_folders/defaults/main.yml [hunk 56]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_5_copy_ui_folders/defaults/main.yml]: The fifth...

File: provisioning/roles/sp_mini_5_copy_ui_folders/tasks/main.yml [hunk 57]
```diff
@@ -0,0 +1,23 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_5_copy_ui_folders/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_5_copy_ui_folders/tasks/main.yml [hunk 58]
```diff
@@ -0,0 +1,23 @@
...
+    state: absent
+
+- name: Copy snowplow-mini.conf under nginx config dir
+  become: yes
+  copy:
+    remote_src: true
+    src: "{{configs_dir}}/snowplow-mini.conf"
+    dest: "/etc/nginx/conf.d"
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_5_copy_ui_folders/tasks/main.yml]: double whitespace

File: provisioning/roles/sp_mini_6_setup_init/defaults/main.yml [hunk 59]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_6_setup_init/defaults/main.yml]: The sixth...

File: provisioning/roles/sp_mini_6_setup_init/tasks/main.yml [hunk 60]
```diff
@@ -0,0 +1,37 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_6_setup_init/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_6_setup_init/tasks/main.yml [hunk 61]
```diff
@@ -0,0 +1,37 @@
...
+    - snowplow_stream_enrich_0.10.0
+    - snowplow_elasticsearch_sink_good_0.8.0
+    - snowplow_elasticsearch_sink_bad_0.8.0
+    - iglu_server_0.2.0
+    - nginx_passenger
+    - nginx
+    - elasticsearch
+    - caddy_init 
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_6_setup_init/tasks/main.yml]: double whitespace

File: provisioning/roles/sp_mini_7_configure/defaults/main.yml [hunk 62]
```diff
@@ -0,0 +1,15 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/defaults/main.yml]: The seventh...

File: provisioning/roles/sp_mini_7_configure/tasks/main.yml [hunk 63]
```diff
@@ -0,0 +1,86 @@
+---
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_7_configure/tasks/main.yml [hunk 64]
```diff
@@ -0,0 +1,86 @@
...
+  pause:
+    seconds: 15
+
+- name: Starting Iglu Server
+  become: yes
+  service: 
+    name: iglu_server_0.2.0 
+    state: started
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_7_configure/tasks/main.yml [hunk 65]
```diff
@@ -0,0 +1,86 @@
...
+- name: Sleep 30 second
+  pause:
+    seconds: 30
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_7_configure/tasks/main.yml [hunk 66]
```diff
@@ -0,0 +1,86 @@
...
+  service: 
+    name: caddy_init
+    state: started
+
+- name: Sleep 30 second
+  pause:
+    seconds: 30
+
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/tasks/main.yml]: double whitespace

File: provisioning/roles/sp_mini_7_configure/tasks/main.yml [hunk 67]
```diff
@@ -0,0 +1,86 @@
...
+- name: curl put good-mapping.json
+  shell: "curl -XPUT 'http://localhost:9200/good' -d @{{es_dir}}/good-mapping.json"
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/tasks/main.yml]: whitespace

File: provisioning/roles/sp_mini_7_configure/tasks/main.yml [hunk 68]
```diff
@@ -0,0 +1,86 @@
...
+- name: curl put bad-mapping.json
+  shell: "curl -XPUT 'http://localhost:9200/bad' -d @{{es_dir}}/bad-mapping.json"
+
+
```
Comment (jbeemster) [inline provisioning/roles/sp_mini_7_configure/tasks/main.yml]: double whitespace

File: provisioning/with_building_ui.yml [hunk 69]
```diff
@@ -0,0 +1,16 @@
+---
+
```
Comment (jbeemster) [inline provisioning/with_building_ui.yml]: whitespace

File: provisioning/without_building_ui.yml [hunk 70]
```diff
@@ -0,0 +1,12 @@
+---
+
```
Comment (jbeemster) [inline provisioning/without_building_ui.yml]: whitespace

File: utils/scripts/user_data.sh [hunk 71]
```diff
@@ -0,0 +1,38 @@
...
+inserted_line=""
+sed -i '1d' /home/ubuntu/snowplow/configs/Caddyfile #delete first line of the default Caddyfile 
+if [[ "${tls_cond}" == "on" ]]; then
+  inserted_line="$domain_name *:80 { \n        tls example@example.com \n"
+else
+  inserted_line="*:80 { \n        tls off \n"
+fi
+sed -i "1s/^/${inserted_line}/" /home/ubuntu/snowplow/configs/Caddyfile
+
+
```
Comment (jbeemster) [inline utils/scripts/user_data.sh]: whitespace

File: utils/scripts/user_data.sh [hunk 72]
```diff
@@ -0,0 +1,38 @@
...
+#add username and password to Caddyfile for basic auth
+sed -i "s/USERNAME_PLACEHOLDER/$username/g" /home/ubuntu/snowplow/configs/Caddyfile
+sed -i "s/PASSWORD_PLACEHOLDER/$password/g" /home/ubuntu/snowplow/configs/Caddyfile
+sudo service caddy_init restart
+
```
Comment (jbeemster) [inline utils/scripts/user_data.sh]: whitespace

File: provisioning/resources/ui/js/components/ControlPlane.tsx (hunk 22 above)
Comment (alexanderdean) [inline provisioning/resources/ui/js/components/ControlPlane.tsx]: Agree

**Comment by jbeemster** [comment]
Other changes needed:

* [ ] https://github.com/snowplow/snowplow-mini/tree/v0.3.0/provisioning/resources/ui - the Quickstart is not correct here and the README is badly formatted.
* [ ] https://github.com/snowplow/snowplow-mini/blob/v0.3.0/provisioning/resources/ui/js/components/Overview.tsx#L52-L57 - these versions have all been changed now and should be reflected in the UI
* [ ] https://github.com/snowplow/snowplow-mini/blob/v0.3.0/provisioning/resources/ui/js/components/Overview.tsx#L61 - the overview image has the collector on Port 8080; the image needs to be updated so that it does not contain this.

**Comment by jbeemster** [comment]
@aldemirenes this is looking good to me.  I think we just need one more QA pass from Nick and it should be good to go!

**Comment by alexanderdean** [comment]
Exciting!

**Comment by jbeemster** [comment]
@alexanderdean green-light from me!

**Comment by alexanderdean** [comment]
Awesome - @aldemirenes please cut the release!

**Pull request merged**